core/tests/ledger_cleanup.rs

```diff
@@ -619,23 +589,22 @@ mod tests {
             insert_timer,
             num_shreds_total as f32 / insert_timer.as_s(),
         );
-        let u1 = storage_previous;
 
-        // Poll on some compaction happening
-        info!("Begin polling for compaction ...");
-        let start_poll = Instant::now();
-        while blockstore.storage_size().unwrap_or(0) >= u1 {
-            if start_poll.elapsed().as_secs() > ROCKSDB_FLUSH_GRACE_PERIOD_SECS {
-                break;
-            }
-            std::thread::sleep(Duration::from_millis(200));
+        // Check if LedgerCleanupService has been purging slots to keep the
+        // blockstore under max_ledger_shreds.
+        // LedgerCleanupService won't do anything unless we have had
+        // >=DEFAULT_PURGE_SLOT_INTERVAL slots inserted, so only check if we
+        // have actually inserted that many slots or more.
+        let last_inserted_slot = benchmark_slots - 1;
+        let cleanup_service_may_have_run = last_inserted_slot > DEFAULT_PURGE_SLOT_INTERVAL;
+        if cleanup_service && cleanup_service_may_have_run {
+            let (exceeded_max_ledger_shreds, _, _) = LedgerCleanupService::find_slots_to_clean(
+                &blockstore,
+                last_inserted_slot - DEFAULT_PURGE_SLOT_INTERVAL,
+                max_ledger_shreds,
+            );
+            assert!(!exceeded_max_ledger_shreds);
         }
-        info!(
-            "Done polling for compaction after {}s",
-            start_poll.elapsed().as_secs_f32()
-        );
-
-        let u2 = storage_previous;
```

Comment:
Let's remove this check in this PR.  As the correct isn't something trivial to implement, let's put it in a separate PR.

Reply:
Done